packages/lage/src/logger/initReporters.ts

```diff
@@ -14,23 +15,25 @@ export function initReporters(config: Config) {
     logLevel = LogLevel[config.logLevel as LogLevelString];
   }
 
-  const reporters: Array<AdoReporter | JsonReporter | NpmLogReporter> = [
-    config.reporter === "json"
-      ? new JsonReporter({ logLevel })
-      : config.reporter === "dgml"
-      ? new DgmlReporter()
-      : new NpmLogReporter({
-          logLevel,
-          grouped: config.grouped,
-          npmLoggerOptions: config.loggerOptions,
-        }),
-  ];
+  let reporters: Array<AdoReporter | JsonReporter | NpmLogReporter | DgmlReporter | CustomReporter> = [];
+  if (config.reporter.includes("json")) {
+    reporters.push(new JsonReporter({ logLevel }));
+  } else if (config.reporter.includes("dgml")) {
+    reporters.push(new DgmlReporter());
+  } else {
+    reporters.push(new NpmLogReporter({ logLevel, grouped: config.grouped, npmLoggerOptions: config.loggerOptions }));
+  }
 
-  if (config.reporter === "adoLog") {
-    // Will always include NpmLogReporter and add AdoReporter
+  // Will always include NpmLogReporter and add AdoReporter
+  if (config.reporter.includes("adoLog")) {
     reporters.push(new AdoReporter());
   }
 
+  // Will always include CustomReporter as well to pass metadata.
+  if (config.reporter.includes("lage-custom-reporter.js")) {
```

Comment:
We should loop through `config.reporters` and add them to the `reporters` so that this config is flexible.

```ts
const configReporters = Array.from(config.reporter);
for (const reporter of configReporters) {
  if (reporter.includes("json")) {
    reporters.push(new JsonReporter({ logLevel }));
  } else if (reporter.includes("dgml")) {
    reporters.push(new DgmlReporter());
  } else if (reporter.match(/\.[jt]s$/)) {
    try {
      const customReporter = require(path.isAbsolute(reporter) ? reporter : path.join(process.cwd(), reporter));
      reporters.push(customReporter);
    } catch (e) {
      throw new Error("custom reporter is invalid");
    }
  } else {
    reporters.push(new NpmLogReporter({ logLevel, grouped: config.grouped, npmLoggerOptions: config.loggerOptions }));
  }
} 
```

Reply:
Made the changes